decoder/src/log/stdout_logger.rs

```diff
@@ -131,50 +131,70 @@ impl StdoutLogger {
 }
 
 /// Printer for `DefmtRecord`s.
-struct Printer<'a> {
+pub(super) struct Printer<'a> {
     record: Record<'a>,
     format: &'a [LogSegment],
     min_timestamp_width: usize,
 }
 
 impl<'a> Printer<'a> {
-    pub fn new(record: Record<'a>, format: &'a [LogSegment]) -> Self {
+    fn new(record: Record<'a>, format: &'a [LogSegment]) -> Self {
         Self {
             record,
             format,
             min_timestamp_width: 0,
         }
     }
 
+    pub fn new_defmt(record: &'a DefmtRecord<'a>, format: &'a [LogSegment]) -> Self {
```

Comment:
What's the purpose of this?

Reply:
To create a new `Printer` from a `DefmtRecord`. We could also use `Printer::new(Record::Defmt(record), format)`, it is just a bit more convenient.